arelle/PackageManager.py

```diff
@@ -444,6 +450,96 @@ def validateTaxonomyPackage(cntlr, filesource, packageFiles=[], errors=[]) -> bo
             errors.append("tpe:metadataFileNotFound")
         return len(errors) == numErrorsOnEntry
 
+def validateReportPackage(filesource, errors=[]) -> bool:
+        numErrorsOnEntry = len(errors)
+        rptPkgExt = os.path.splitext(filesource.baseurl)[1]
+        rptPkgFile = filesource.reportPackageFile
+        rptPkgObj = {"documentInfo":{"documentType":"https://xbrl.org/report-package/2023"}} #default doc type
+        def loadDict(keyValuePairs):
+            _dict = {}
+            for key, value in keyValuePairs:
+                if key not in _dict:
+                    _dict[key] = value
+                else:
+                    filesource.cntlr.addToLog(_("JSON duplicated key %(key)s"),
+                                   messageCode="rpe:invalidJSON",
+                                   file=os.path.basename(filesource.baseurl),
+                                   messageArgs={"key": key},
+                                   level=logging.ERROR)
+                    errors.append("rpe:invalidJSON")
+            return _dict
+        if rptPkgFile:
+            pkgFilePath = f"{filesource.basefile}/{rptPkgFile}"
+            _file = filesource.file(pkgFilePath, binary=True)[0]
+            bytes = _file.read(16) # test encoding
+            m = EBCDIC_Bytes_Pattern.match(bytes)
+            if m and not NEVER_EBCDIC_Bytes_Pattern.findall(bytes):
```

Comment:
Out of curiosity, why are these checks for EBCDIC/UTF-7 needed?  (And how do the regex ranges work?)